src/main/scala/com.snowplowanalytics/iglu.schemaddl/migrations/Migration.scala

```diff
@@ -98,26 +106,102 @@ object Migration {
     } yield cell
   }
 
-  def buildMigrationMatrix(schemas: NonEmptyList[IgluSchema]): NonEmptyList[OrderedSchemas] =
+  /**
+    * Creates model groups of schemas and returns them as OrderedSchemas
+    * @param schemas List of schema to create migration matrix
+    * @return List of IgluSchema as Ior.left if some groups in schemas contain
+    *         only one schema and no meaningful migrations can be provided
+    *         or List of OrderedSchemas as Ior.right if there are more than
+    *         one schemas in some groups and migrations can be provided
+    */
+  def buildMigrationMatrix(schemas: NonEmptyList[IgluSchema]): Ior[NonEmptyList[IgluSchema], NonEmptyList[OrderedSchemas]] = {
     schemas
       .groupByNem { case SelfDescribingSchema(SchemaMap(SchemaKey(v, n, _, SchemaVer.Full(model, _, _))), _) => (v, n, model) }
       .toNel
-      .flatMap { case (_, s) =>
-        val matrix = buildMatrix(s.toList).map { case (_, _, ss) => OrderedSchemas(ss) }
-        NonEmptyList.fromListUnsafe(matrix)
-      }
+      .nonEmptyPartition { case (_, s) =>
+        buildMatrix(s.toList).map { case (_, _, ss) => OrderedSchemas(ss) } match {
+          case Nil => Left(s)
```

Comment:
If `buildMatrix` returns `Nil` then `s` contains only one element. We can write `s.head` in order to avoid flattening iit later.